index.html

```diff
@@ -4,185 +4,222 @@
-];
-
-var types={
-"Option":          {"url":"https://doc.rust-lang.org/stable/std/option/enum.Option.html",  "ERRNAME":"None"},
-"Result":          {"url":"https://doc.rust-lang.org/stable/std/result/enum.Result.html",  "ERRNAME":"the error"},
-"Iterator<Option>":{"url":"https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html", "ERRNAME":"None"},
-"Iterator<Result>":{"url":"https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html", "ERRNAME":"error"},
-};
-
-function getColumn(key,constraints) {
-	var selected=methods.filter(function(el){
-		for (c in constraints)
-			if (el[c]!=constraints[c])
-				return false;
-		return true;
-	});
-	return _(selected).map(function(row){
-		return row[key];
-	}).sort().uniq(true).value();
-}
-
-function makeOption(text,errname,selected) {
-	var opt=$('<option>').val(text).text(text.replace("$ERRNAME",errname,"g"));
-	if (text==selected)
-		opt.attr('selected',true);
-	return opt;
-}
-
-function mod_for(maybe_mod) {
-	if (_.endsWith(maybe_mod,"_mod")) {
-		return maybe_mod.replace(/_mod$/,"");
-	} else {
-		return null;
-	}
-}
-
-//"have":[],"want":[],"by":[],"by_mod":[],"or":[],"or_mod"
-function showColumns(selected) {
-	["have","want","by","by_mod","or","or_mod"].forEach(function(k){
-		var opts=getColumn(k,_.omit(selected,k));
-		var select=$('#'+k);
-		select.empty();
-		if (_.eq(opts,[null]) || mod_for(k) && !selected.hasOwnProperty(mod_for(k))) {
-			select.hide();
-			select.val(null);
-		} else {
-			select.show();
-			var matched=false;
-			var opts=_.compact(opts).map(function(row){
-				var this_opt={};
-				this_opt[k]=row;
-				var errname=_(getColumn('method',_.merge(this_opt,selected))).map(function(method){
-					return types[method[0]].ERRNAME;
-				}).uniq().join('/');
-				matched=matched||(row==selected[k]);
-				return makeOption(row,errname,selected[k]);
+	<div class="action">
+		<label>I have</label>
+		<select id="have"></select>
+	</div>
+	<div class="action">
+		<label>I want</label>
+		<select id="want"></select>
+	</div>
+	<div class="action">
+		<label>By</label>
+		<select id="by"></select>
+	</div>
+	<div class="action">
+		<label>&hellip;</label>
+		<select id="by_mod"></select>
+	</div>
+	<div class="action or">
+		<label>Or</label>
+		<select id="or"></select>
+	</div>
+	<div class="action">
+		<label>&hellip;</label>
+		<select id="or_mod"></select>
+	</div>
+
+	<div class="clear">
+		<button id="clear_all">Reset all</button> &middot;
+		<a href="https://github.com/jethrogb/rust-combinators">source</a>
+	</div>
+	<div class="clear result">
+		<code id="disp"></code>
+	</div>
+
+	<script type="text/javascript">
+		var methods = [
+			{ "method": ["Option", "expect"], "have": "Option<T>", "want": "T", "by": "propagating the result", "by_mod": null, "or": "panicking", "or_mod": "with a specific error message" },
```

Comment:
Can you keep the original formatting here?

Reply:
Let me just revert reformatting altogether

Reply:
Force pushed with reverted formatting.